frontend/packages/dev-console/src/components/pipelines/modals/common/ModalStructure.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { ModalBody, ModalSubmitFooter, ModalTitle } from '@console/internal/components/factory';
+
+type ModalStructureProps = FormikProps<FormikValues> & {
+  children: React.ReactNode;
+  close: () => void;
```

Comment:
Instead of we defining this, we could use the existing type `ModalComponentProps` it has close and cancel functions defined there.

Reply:
Hmm... an interesting point. Maybe I can do `Partial<ModalComponentProps>` and that way at least it's not going to have any required prop issues (if there are any).

Reply:
No there isn't any, both are optional params.